generatorcore/industry2018/energy_sum.py

```diff
@@ -1,7 +1,26 @@
-from dataclasses import dataclass
+# pyright: strict
+from dataclasses import InitVar, dataclass
+from generatorcore.utils import div
 
 
 @dataclass
 class EnergySum:
     # Used by s_fossil, s_renew
     energy: float
+
+
+@dataclass
+class Empty:
+    # Used s_renew_hydrogen, s_renew_emethan
+    pass
+
+
+@dataclass
+class Energy_pct:
```

Comment:
@Ultramarin95, @bgrundmann:
Die Klasse scheint mir auch relativ sinnlos zu sein. Hier werden jeweils die Prozente ausgerechnet, aber soweit ich erkennen kann, werden die Prozente niemals irgendwo angewendet/ausgewertet.

Oder seht ihr darin irgendeinen Zweck? Das könnte schon in Zukunft mal interessant sein, aber aktuell ist es quasi toter Code. Wollt ihr den drin lassen oder soll ich hier überall 'nur' EnergySum verwenden?

Das gleiche Verhalten scheint sich durch die meisten Sektoren durchzuziehen.

Die Prozente könnten irgendwann interessant sein, aber in der usage.json tauchen sie auch nicht auf.

Wird die usage.json eigentlich generiert oder woher kommt die komische Sortierung? Das wäre auch mal eine Überlegung, die nach Sektoren zu sortieren, damit man schneller einen Überblick bekommt ;)

Reply:
Ich nehme an die Zahlen kommen daher das die in der excel sozusagen teil der "UI" waren. Denn dafür sind die natürlich nützlich. Im prinzip haben wir im python generator glaube ich nie wirklich darüber nachgedacht welche zahlen der generator noch berechnen sollte und welche so trivial sind das es der job von einem zweiten schritt seien sollte.

Ich glaube es wäre unter umständen eine gute idee alle prozente rauszuschmeissen und dann die paar die wir wirklich in knud brauchen, separat in knud zu berechnen. Ähnliches gilt vielleicht auch für einige der aggregationen.

Bin mir aber nicht sicher ob wir das nicht lieber in nem separaten PR machen sollten.

Was usage.json angeht: record_generator_usage.py in knud (das war der zugehörige merge request https://code.germanzero.org/germanzero/knud/-/merge_requests/69 -- usprünglich habe ich das selbe über ne unix pipeline (`egrep -H -o ... | sort | uniq` oder so ähnlich). 

Die Idee war übrigens gar nicht das die erzeugte Datei möglichst lesbar sein sollte. 

Für schnell drin suchen reicht grep oder die suchfunktion des editors-of-your-choice. Und wenn man zum Beispiel alle pct über alle sektoren sehen will ist auch wieder grep und vielleicht jq oder so die tools der wahl. Und ne Sortierung nach sektor gar nicht unbedingt die nützlichste sortierung. Das heisst nicht das ich dagegen bin, die Datei zu sortieren.  Sondern nur das ich bis jetzt keine Sortierung vermisst habe, wenn ich die Datei gebraucht habe.

Reply:
Prozente:
Ich bin mir auch nicht sicher, ob wir sie rausschmeißen sollten. Wenn wir sie nicht brauchen, würde ich sie rausnehmen, aber wenn wir sie brauchen, weiß ich nicht, ob ich es so gut finde, die in KNUD zu berechnen. Ich finde die Berechnungen sollten gebündelt im Generator liegen und man müsste sonst immer auch erstmal schauen, welche Prozente man wovon eigentlich berechnet ... Naja, wie du sagst, das ist ein Thema für später und sollte separat diskutiert werden 

usage.json
D.h. die Datei wurde einmal generiert und wird jetzt eher manuell angepasst, wenn ich das richtig verstanden habe? Naja, ich fände eine Sortierung schon hilfreich. Aber danke für die Infos ... und ja, auch wieder ein separates Thema für später ...